File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* findNode(ListNode* temp,int n){
        int cnt=1;
        while(temp->next!=NULL){
            if(cnt==n) return temp;
            cnt++;
            temp=temp->next;
        }
        return temp;
    }
    ListNode* rotateRight(ListNode* head, int k) {
        if (k==0 || head==NULL){
            return head;
        }
        
        int len=1;
        ListNode* tail=head;
        while(tail->next!=NULL){
            len++;
            tail=tail->next;
        }
        if(k%len==0) return head;
        k=k%len;
        tail->next=head;
        ListNode* newNode=findNode(head,len-k);
        head=newNode->next;
        newNode->next=NULL;
        return head;
    }
};

File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    ListNode findNode(ListNode head,int n){
        int cnt=1;
        while(head.next!=null){
            if(cnt==n) return head;
            cnt++;
             head= head.next;
        }
        return  head;
    }
    public ListNode rotateRight(ListNode head, int k) {
        ListNode tail=head;
        if(head==null || head.next==null) return head;
        
        int n=1;
        while(tail.next!=null){
            tail=tail.next;
            n++;
        }
        if(k%n==0 ) return head;
        k=k%n;
        tail.next=head;
        ListNode newNode= findNode(head,n-k);
        head=newNode.next;
        newNode.next=null;
        return head;
    }
}